File: javascript/messagesDapp/sendMessage.js
import { PWRWallet } from "@pwrjs/core";
import dotenv from 'dotenv';
dotenv.config();

// Setting up your wallet in the SDK
const privateKey = process.env.PRIVATE_KEY;
const wallet = new PWRWallet(privateKey);

async function sendMessage() {
    const obj = { message: "Hello World!" };
    const data = Buffer.from(JSON.stringify(obj), 'utf8');
    const vmId = 1234;

    const res = await wallet.sendVMDataTxn(vmId, data);
    console.log(res.transactionHash);
}
sendMessage();

File: go/messagesDapp/sendMessage.go
package messagesDapp

import (
	"fmt"
	"os"
	"encoding/json"
	"github.com/joho/godotenv"
	"github.com/pwrlabs/pwrgo/wallet"
)

// var PrivateKey = "0x9D4428C6E0638331B4866B70C831F8BA51C11B031F4B55EED4087BBB8EF0151F"

func SendMessage() {
	// Setting up your wallet in the SDK
	godotenv.Load()
	privateKey := os.Getenv("PRIVATE_KEY")
	wallet := wallet.FromPrivateKey(privateKey)

	vmId := 123
	data, _ := json.Marshal(map[string]string{"message": "Hello World!"})

	tx := wallet.SendVMData(vmId, data)

	if tx.Success {
		fmt.Printf("Transaction Hash: %s\n", tx.TxHash)
	} else {
		fmt.Printf("Failed to send transaction: %s\n", tx.Error)
	}
}
